actions/archive2disk/v1/pkg/archive/utils.go

```diff
@@ -0,0 +1,107 @@
+package archive
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+	// See https://github.com/opencontainers/go-digest#usage
+	_ "crypto/sha256"
+	_ "crypto/sha512"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func extractTarDirectory(root string, checksum string, r io.Reader) error {
+	var verifier digest.Verifier
+	if checksum != "" {
+		log.Infof("checksum validation during untar [%s]", checksum)		
+		digest, err := digest.Parse(checksum)
+		if err != nil {
+			return errors.New("failed to parse digest")
```

Comment:
Can you include the `digest.Parse` error in the returned error? (e.g., `fmt.Errorf("failed to parse digest - %w", err)`)

Reply:
I made the digest.Parse change.  For the CHECKSUM requirement,  I did consider this but wanted to minimize backward compatibility issues, so didn't want to enforce it as a requirement.   I'm good either way if others want to provide input.

Reply:
We discussed this in the [community meeting today](https://docs.google.com/document/d/1cEObfvQ9Tdp8zIIIg9O7P5i3CKaSj2t3JTxEufDxwWs/edit?usp=sharing) and the consensus was having the checksum be opt-out was preferred.